lib/fog/vsphere/requests/compute/get_vm_first_sata_controller.rb

```diff
@@ -0,0 +1,20 @@
+module Fog
+  module Vsphere
+    class Compute
+      class Real
+        def get_vm_first_sata_controller(vm_id)
+          ctrl = get_vm_ref(vm_id).config.hardware.device.find { |device| device.is_a?(RbVmomi::VIM::VirtualAHCIController) }
```

Comment:
Just reading over this: can't this return `nil`? If so, the rest of the code would crash.

Reply:
+1 to using ruby safe navigation operator. I have not looked at the code yet, I had to redeploy vcenter 8, so will test this this week. But yeah this needs a fix

Reply:
Don't you just want to return `nil` if it's not found instead?

Reply:
Updated the code to raise `Not Found ` error if it's nil based on other compute/requests.